src/main/java/org/springframework/data/tarantool/core/BaseTarantoolTemplate.java

```diff
@@ -350,8 +356,8 @@ protected <T> ValueConverter<Value, List<T>> getListValueConverter(Class<T> enti
                 .collect(Collectors.toList());
     }
 
-    protected <T> CallResultMapper<TarantoolResult<TarantoolTuple>,
-            SingleValueCallResult<TarantoolResult<TarantoolTuple>>>
+    protected <T> CallResultMapper<TarantoolTupleResultImpl,
```

Comment:
Using a concrete `Impl` generic type in the function return type is not correct. Why do you want to change this?

We can use the concrete class as a parameter of the factory method anyway.

Reply:
We have to pass class as argument of method. And I didn't want to use ugly casts (Class<TarantoolResult>) (Class<?>) anyClass